src/allocator/epoch_zone_heap.cc

```diff
@@ -1187,6 +1222,52 @@ void *EpochZoneHeap::GlobalToLocal(GlobalPtr global_ptr) {
     return local_ptr;
 }
 
+void EpochZoneHeap::delayed_free_fn() {
+    TRACE();
+    ASSERT_IS_OPEN();
+    Offset offset;
+
+    for (int shelf_num = 0; shelf_num < total_mapped_shelfs_; shelf_num++) {
+        if (fam_atomic_u64_read(&gh_->destroy_in_progress)) {
+            is_invalid_ = true;
+            for (int shelf_num = 0; shelf_num < total_mapped_shelfs_;
+                shelf_num++) {
+                rmb_[shelf_num]->MarkInvalid();
+            }
+                Close();
+                return;
+        }
+            uint64_t i = 0;
+            EpochManager *em = EpochManager::GetInstance();
+            for (; i < kFreeCnt; i++) {
+
+              {
+                EpochOp op(em);
+                EpochCounter e = op.reported_epoch();
+                offset = global_list_[shelf_num][(e) % kListCnt]
+                             .pop(bitmap_start_[shelf_num]) *
+                         min_obj_size_;
+              }
+                if (offset == 0)
+                    break;
+                LOG(trace) << " freeing block [" << offset << "]";
+		rmb_[shelf_num]->ClearData(offset);
```

Comment:
we might have to run git-clang-format for the first commit. We can do a soft reset and do for everything.